### DISCUSSION TITLE
Marking detect object with Custom number

### DISCUSSION THREAD

**Discussion by aleem75321** [ultralytics/ultralytics#5684] [Q&A]
I have a single class and no ID and saving detects objects. I am Saving xywhn in a text file. So How do I identify which object's location is? I am saving the annotated image

**Comment by glenn-jocher** [comment]
@aleem75321 hello! If you're saving detections with xywhn format in a text file and also saving annotated images, each line in your text file corresponds to a detected object in the image. The objects are not inherently numbered, but they are listed in the order they are detected. 

To identify which object's location corresponds to which detection in your annotated image, you could modify the code that generates the annotated images to include a unique identifier (such as a number) next to each bounding box. This way, you can cross-reference the numbered detections in the image with the lines in your text file.

Remember, the text file saves the detections in the format `[class x_center y_center width height]`, normalized by image width and height. The first value is the class (which in your case will always be the same since you have a single class), followed by the bounding box attributes.

If you need further guidance on how to modify the code for your specific use case, please refer to the documentation or consider posting a detailed question on the GitHub issues page with what you've tried and where you're getting stuck. Happy coding! 😊
